includes/class-base-widget.php

```diff
@@ -397,9 +397,126 @@ protected function render_form_textarea( array $field ) {
+		$hidden_text_field    = ( ! $hours['custom_text_checkbox'] ) ? ' style="display:none;"' : '';
+		$custom_text_field    = '<input type="text" name="' . $field['name'] . '[custom_text]" id="' . $field['name'] . '[custom_text]" class="widefat custom_text_field" value="' . $hours['custom_text'] . '"' . $hidden_text_field . '/>';
+		$closed_checkbox      = '<input name="' . $field['name'] . '[not_open]" id="' . $field['name'] . '[not_open]" class="js_wpcw_closed_checkbox" type="checkbox" value="1" ' . $this->checked( $hours['not_open'], true ) . '><label for="' . $field['name'] . '[not_open]" class="js_wpcw_closed_checkbox"><small>' . esc_html__( 'Closed', 'contact-widgets' ) . '</small></label>';
+
+		printf(
+			'<div class="day-container">%1$s<p>%2$s %3$s</p></div>',
+			'<strong>' . esc_html( ucwords( $day ) ) . '<span class="day-checkbox-toggle">' . $apply_to_all_toggle . $custom_text_checkbox . $closed_checkbox . '</strong></span>',
+			$this->render_hours_selection( $field, sanitize_title( $day ), $hours ),
+			$custom_text_field
+		);
+
+	}
+
+	/**
+	 * Render the 'Hours' select fields
+	 *
+	 * @param array $field Field data
+	 * @param array $day   The current day in the iteration.
+	 * @param array $hours The array of times.
+	 *
+	 * @since NEXT
+	 *
+	 * @return mixed
+	 */
+	protected function render_hours_selection( $field, $day, $hours ) {
+
+		ob_start();
+
+		$times = $this->get_half_hour_time_array();
+
+		$field['name'] = str_replace( 'value', strtolower( $day ), $field['name'] );
+
+		$disabled_field = $field['disabled'] ? ' disabled="disabled"' : '';
+		$hidden_field   = $hours['custom_text_checkbox'] ? ' style="display:none;"' : '';
+
+		?>
+
+		<select name="<?php echo esc_attr( $field['name'] . '[open]' ); ?>" <?php echo $disabled_field . $hidden_field; ?>>
+
+		<?php
+
+		foreach ( $times as $time ) {
+
+			?>
+
+			<option <?php selected( $hours['open'], $time ); ?>><?php echo esc_html( $time ); ?></option>
+
+			<?php
+
+		}
+
+		?>
+
+		</select>
+
+		<select name="<?php echo esc_attr( $field['name'] . '[closed]' ); ?>" <?php echo $disabled_field . $hidden_field; ?>>
+
+		<?php
+
+		foreach ( $times as $time ) {
+
+			?>
+
+			<option <?php selected( $hours['closed'], $time ); ?>><?php echo esc_html( $time ); ?></option>
+
+			<?php
+
+		}
+
+		?>
+
+		</select>
+
+		<?php
+
+		return ob_get_clean();
+
+	}
+
+	/**
+	 * Generate an array of times in half hour increments
+	 *
+	 * @since NEXT
+	 *
+	 * @return array
+	 */
+	protected function get_half_hour_time_array() {
+
+		$half_hour_steps = range( 0, 47 * 1800, 1800 );
```

Comment:
@jonathanbardo @fjarrett What do you think about introducing a filter here to allow users to set increments to whole hours, 15 mins, 10 mins etc.? Do you think that would be beneficial?